bookwyrm/tests/actions/test_readthrough.py

```diff
@@ -0,0 +1,74 @@
+from django.test import TestCase, Client
+from django.utils import timezone
+from datetime import datetime
+
+from bookwyrm import models
+
+class ReadThrough(TestCase):
+    def setUp(self):
+        self.client = Client()
+
+        self.work = models.Work.objects.create(
+            title='Example Work'
+        )
+
+        self.edition = models.Edition.objects.create(
+            title='Example Edition',
+            parent_work=self.work
+        )
+        self.work.default_edition = self.edition
+        self.work.save()
+
+        self.user = models.User.objects.create_user(
+            'cinco', 'cinco@example.com', 'seissiete',
+            local=True, localname='cinco')
+
+        self.client.force_login(self.user)
+
+    def test_create_basic_readthrough(self):
+        """A basic readthrough doesn't create a progress update"""
+        self.assertEqual(self.edition.readthrough_set.count(), 0)
+
+        self.client.post('/start-reading/{}'.format(self.edition.id), {
+            'start_date': '2020-11-27',
```

Comment:
this might have the side effect of creating a "started reading" status and then triggering a `broadcast` task, which could be why the tests are hanging. you can use `with patch('bookwyrm.broadcast.broadcast_task.delay')` if that's what's happening. Stopping celery and redis when you run the tests will show these errors locally. I have learned the hard way!

Reply:
Oh rad, that's very helpful, thanks! I'll take a peek at that, hopefully the logs will confirm once they time out again in a couple hours :upside_down_face: